File: Laporan 3 (WEEK 9) Final Tugas Besar (Mobile App)/lib/models/book.dart
class Book {
  final String id;
  final String title;
  final String author;
  final String description;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.description,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      author: json['author'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'author': author,
      'description': description,
    };
  }
}

File: webfix/models/book.ts
export interface Book {
  id: string
  title: string
  author: string
  coverUrl: string
  description: string
  category: string
  stock: number
}

export interface BorrowingItem {
  id: string
  bookId: string
  userId: string
  title?: string
  borrowDate: Date
  returnDueDate: Date
  isReturned: boolean
  returnedAt: Date | null
  daysLate?: number
  damageCount?: number
  damageNote?: string
  fine?: number
}

export interface CartItem {
  id: string
  bookId: string
  title?: string
  author?: string
  coverUrl?: string
  quantity: number
  addedAt: Date
  stock?: number
}
